apis/reaper/v1alpha1/reaper_types.go

```diff
@@ -149,33 +175,12 @@ type AutoScheduling struct {
 }
 
 type ReaperClusterTemplate struct {
-	ReaperDatacenterTemplate `json:",inline"`
+	ReaperTemplate `json:",inline"`
 
-	// The keyspace to use to store Reaper's state. Will default to "reaper_db" if unspecified. Will be created if it
-	// does not exist, and if this Reaper resource is managed by K8ssandra.
-	// +kubebuilder:default="reaper_db"
 	// +optional
-	Keyspace string `json:"keyspace,omitempty"`
-
-	// Defines the username and password that Reaper will use to authenticate CQL connections to Cassandra clusters.
-	// These credentials will be automatically turned into CQL roles by cass-operator when bootstrapping the datacenter,
-	// then passed to the Reaper instance, so that it can authenticate against nodes in the datacenter using CQL. If CQL
-	// authentication is not required, leave this field empty. The secret must be in the same namespace as Reaper itself
-	// and must contain two keys: "username" and "password".
-	// +optional
-	CassandraUserSecretRef corev1.LocalObjectReference `json:"cassandraUserSecretRef,omitempty"`
-
-	// Defines the username and password that Reaper will use to authenticate JMX connections to Cassandra clusters.
-	// These credentials will be automatically passed to each Cassandra node in the datacenter, as well as to the Reaper
-	// instance, so that the latter can authenticate against the former. If JMX authentication is not required, leave
-	// this field empty. The secret must be in the same namespace as Reaper itself and must contain two keys: "username"
-	// and "password".
-	// +optional
-	JmxUserSecretRef corev1.LocalObjectReference `json:"jmxUserSecretRef,omitempty"`
-
-	// Defines the secret which contains the username and password for the Reaper UI and REST API authentication.
-	// +optional
-	UiUserSecretRef corev1.LocalObjectReference `json:"uiUserSecretRef,omitempty"`
+	// +kubebuilder:default="PER_DC"
+	// +kubebuilder:validation:Enum:=PER_DC;SINGLE
+	DeploymentMode string `json:"deploymentMode,omitempty"`
```

Comment:
This does not offer users the ability to chose the DC where Reaper is going to be deployed in SINGLE mode. We could add a `SingleDeploymentDcRef` field for that.

Reply:
Is it going to be the first DC each time? It could probably be enough this way as long as we can predict which DC is going to get Reaper.
Also, what will happen if the DC that has Reaper gets decommissioned?

Reply:
> Is it going to be the first DC each time?

Yes currently it is always the first DC in the `kc.Spec.Cassandra.Datacenters` list that gets Reaper deployed.

> what will happen if the DC that has Reaper gets decommissioned?

We don't have that feature yet, but presumably, the DC will be removed from the list, another DC will become the "first" one in the list, and the controller will detect that Reaper should be deployed there, and will make the necessary changes.